core/src/main/java/org/elasticsearch/transport/local/LocalTransport.java

```diff
@@ -288,6 +294,9 @@ private void handleRequest(StreamInput stream, long requestId, LocalTransport so
             if (reg == null) {
                 throw new ActionNotFoundTransportException("Action [" + action + "] not found");
             }
+            // reserve here in order be able to send a response in case we break the limit
+            inFlightRequestsBreaker.addEstimateBytesAndMaybeBreak(messageLengthBytes, "Node in-flight requests");
+            transportChannel.setReservedBytes(messageLengthBytes);
```

Comment:
Well, in case the circuit breaker throws an exception in the constructor of the channel, we don't have any place where we can send an error response to the client (see the catch block of this method). If that is fine I can change that.